stock_voucher/models/stock_picking.py

```diff
@@ -53,11 +51,12 @@ class StockPicking(models.Model):
         readonly=True,
     )
 
-    @api.multi
     @api.depends('voucher_ids.display_name')
     def _compute_vouchers(self):
         for rec in self:
-            rec.vouchers = ', '.join(rec.mapped('voucher_ids.display_name'))
+            rec.update({
+                'vouchers': ', '.join(rec.mapped('voucher_ids.display_name'))
+            })
 
     @api.multi
     def get_estimated_number_of_pages(self):
```

Comment:
This method can be improved:

1. Not need tu import ceil() function:  We can achieve the same result like this, using **//** operator and negating the sign of the operation.

```python
estimated_number_of_pages = int(-(-float(operations) // float(lines_per_voucher)))

>>> 102.0/10.0
10.2
>>> -(-102.0//10.0)
11.0
```
2. Use early return to reduce the complexity of the method

```python
def get_estimated_number_of_pages(self):
    self.ensure_one()
    res = 1 
    lines_per_voucher = self.book_id.lines_per_voucher
    if lines_per_voucher == 0:
        return res

    operations = len(self.move_lines)
    res = int(-(-float(operations) / float(lines_per_voucher)))
    return res
```